### PR TITLE
[Impeller] Fix issue about saveLayer ignoring opacity of paint with advanced blend mode

### PR DESCRIPTION
**Pull request by ColdPaleLight**
fix https://github.com/flutter/flutter/issues/126532

without patch
56V3vfllrj

with patch
StzXmubqcV

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.

### CODE REVIEW

File: impeller/display_list/dl_unittests.cc [hunk 1]
```diff
@@ -447,6 +447,21 @@ TEST_P(DisplayListTest, IgnoreMaskFilterWhenSavingLayer) {
   ASSERT_TRUE(OpenPlaygroundHere(builder.Build()));
 }
 
+TEST_P(DisplayListTest, SaveLayerWithOpacityAndAdvancedBlendMode) {
```
Comment (jonahwilliams) [inline impeller/display_list/dl_unittests.cc:450]: If you make this an Aiks unittest I believe we can get a golden test image out of it so it gets verified on CI.

Comment (jonahwilliams) [reply impeller/display_list/dl_unittests.cc:450]: actually, that would only show an issue if it were an arm64 mac. Still, might be good to do.

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: impeller/display_list/dl_unittests.cc (hunk 1 above)
Comment (ColdPaleLight) [reply impeller/display_list/dl_unittests.cc]: > If you make this an Aiks unittest I believe we can get a golden test image out of it so it gets verified on CI.

Sounds good, done.

**Comment by ColdPaleLight** [comment]
@jonahwilliams I moved the test from dl_unittests to aiks_unittests, and verified locally that there are no issues, but the bot added the label "will affect goldens" to this PR, I don't know how to fix it, would you mind teaching me how to do it? Thanks

**Comment by jonahwilliams** [comment]
I'll do it for now, If we don't have instructions we should write some up

**Comment by jonahwilliams** [comment]
It looks like this is working correctly on arm macbooks and iOS, but on intel or vulkan/gles where framebuffer blending isn't used the opacity still doesn't look correct:

image

**Comment by jonahwilliams** [comment]
It looks like blend_filter_contents.cc doesn't have a path to accept the src snapshot opacity.

**Comment by bdero** [comment]
Patch LGTM.

**Comment by ColdPaleLight** [comment]
@jonahwilliams Thanks for your patch! I added the patch to this PR. Now still need to approve new screenshots on skia gold, since I don't have access, would you mind approving it?

**Review by bdero — APPROVED** [review]
I approved the goldens. LGTM, thanks!

**Comment by chinmaygarde** [comment]
Can we cherry-pick this into stable please? [The request template is here](https://github.com/flutter/flutter/issues/new?assignees=itsjustkevin%2Ccaseyhillers&labels=cp%3A+review&projects=&template=7_cherry_pick.yml&title=%5BCP%5D+%3Ctitle%3E).

**Pull request merged**